.github/workflows/tests_branch.yml

```diff
@@ -2,19 +2,14 @@ name: Branch tests
 
 on:
   push:
-    branches: ["main"]
-  pull_request:
-    branches: ["main"]
   workflow_dispatch:
 
 jobs:
   branch_lint:
-    name: Linting tests
+    name: Linting tests (python files)
     runs-on: ubuntu-20.04
     steps:
-      - name: checkout
-        uses: actions/checkout@v3
-
+      - uses: actions/checkout@v3
       - name: setup python
         uses: actions/setup-python@v4.5.0
```

Comment:
I am a fan of the pytest for linting and formatting, think its a really nice addition. It might be a good maintenance task for DE in the future